### ISSUE TITLE
Expected barcode count does not change with large sample sizes

### ISSUE THREAD

**Issue by slsevilla** [RBL-NCI/iCLIP#139]
Problem: Yuanyuan recently ran a dataset with 12 samples (6 positive controls, 6 negative controls). The current barcode QC schema currently reviews the top 10 observed barcodes only, and would miss 2 barcodes in this instance. Because of the distribution of her dataset (pos/neg controls), however, this is not the case and 3 barcodes are missing. 

![Screen Shot 2023-01-10 at 12 23 27](https://user-images.githubusercontent.com/20726305/211639800-59c3cc83-fc9c-40a1-8c2a-1c003fc4daf7.png)

Solution: 
02_barcode_qc.R should adjust the top number of barcodes observed to match the dataset. 

```
if length of bc_exp < 10 then number_of_observed_barcodes = 10
else if length of bc_exp > 10 and bc_exp < 20 then number_of_observed_barcodes = 20
else if length of bc_exp > 20 then number_of_observed_barcodes = bc_exp + 10
```

I will test this with the current data set and implement the new code to v2.1.1

**Comment by slsevilla** [comment]
After updating code to expand the expected number of barcodes dependent on the number of samples, code is producing results as expected.

The top 20 barcodes, in this case, includes the 12 barcodes that were expected.
<img width="355" alt="Screen Shot 2023-01-15 at 20 55 52" src="https://user-images.githubusercontent.com/20726305/212582543-402a8a36-1cca-472a-86d5-da4049fb2748.png">

Code will be implemented into v2.1.1 of the pipeline with the addition of the passing qc barcode manifest being required for further processing.

**Comment by slsevilla** [comment]
completed with commit a22bd9da48e0e0fff3f0bcb47c0c98e70cc4f4cd
